lib/Test/MockFile.pm

```diff
@@ -1141,7 +1141,31 @@ sub _abs_path_to_file {
 
     return q[/] if $path eq q[/..];
 
-    return $path if $path =~ m{^/};
+    return $path if $path =~ m{^/}xms;
+
+    # ~
+    # ~/...
+    # ~sawyer
+    if ( $path =~ m{ ^(~ ([^/]+)? ) }xms ) {
+        my $username = $2 || getpwuid($<);
+        my $homedir;
+        endpwent;
+        while ( my @pwdata = getpwent ) {
+            if ( $pwdata[0] eq $username ) {
+                $homedir = $pwdata[7];
+                endpwent;
```

Comment:
I've never seen this called???

Reply:
when you call `getpwent` you're looking over it, akin to `each %hash`. You need to reset the iterator with `endpwent`. I'm resetting it in the beginning to make sure I'm starting from the top.